cypress/integration/DeleteOpeningPeriod.spec.ts

```diff
@@ -0,0 +1,51 @@
+/// <reference types="cypress" />
+/// <reference path="../index.d.ts" />
+
+describe('User deletes an opening period', () => {
+  let dataPeriodId: string | undefined;
+  const resourceId = Cypress.env('resource-id');
+  const datePeriodTitlePrefix = 'e2e-test Poistamisen testijakson otsikko';
+  const newTitle = `${datePeriodTitlePrefix} ${new Date().toJSON()}`;
+  const startDate = new Date();
+  const endDate = new Date();
+  endDate.setDate(new Date().getDate() + 1);
+
+  before(() => {
+    cy.createDatePeriod({
+      name: newTitle,
+      startDate,
+      endDate,
+      resourceId,
+    }).then((id: string) => {
+      if (id) {
+        dataPeriodId = id;
+      }
+    });
+  });
+
+  beforeEach(() => {
+    cy.visitResourcePageAsAuthenticatedUser(resourceId);
+  });
+
+  it('User successfully edits an opening period', () => {
+    // Enter in delete modal by clicking the delete-button
+    cy.get('[data-test=resource-opening-periods-list]', {
+      timeout: 5000,
+    })
+      .find(`[data-test="openingPeriodDeleteLink-${dataPeriodId}"]`)
+      .click({ log: true });
+
+    // Check that delete modal exists
+    cy.get('[data-test=modalTitle]', {
+      timeout: 5000,
+    }).should('have.text', 'Oletko varma että haluat poistaa aukiolojakson?');
```

Comment:
I feel like this might be unnecessarily strict check? If we wish to change the text, then this test will fail, even our application will still work ok. I think it would be better to just verify the existence of the element with that data-test id, and leave it at that probably. If there's a more specific element we would want to test, then maybe we could give the data-test attribute to that element?

Reply:
I could just see if the title exists 🤔